common/web/lm-worker/src/main/correction/context-tracker.ts

```diff
@@ -397,10 +447,31 @@ export class ContextTracker extends CircularArray<TrackedContextState> {
           const token = state.tokens[i - poppedTokenCount];
           const matchToken = matchState.tokens[i];
 
+          // TODO:  I'm beginning to believe that searchSpace should (eventually) be tracked
+          // on the tokens, rather than on the overall 'state'.
+          // - Reason:  phrase-level corrections / predictions would likely need a search-state
+          //   across per potentially-affected token.
+          // - Shifting the paradigm should be a separate work unit than the
+          //   context-tracker rework currently being done, though.
```

Comment:
These are notes toward potential future work - things that would need to be addressed to support phrase-level corrections.

Reply:
Added as part of a new issue I'm writing up in relation to this - #12010.